### ISSUE TITLE
PR #6695 Introduces Breaking Change for Flows Outside of Current Directory

### ISSUE THREAD

**Issue by ColeMurray** [PrefectHQ/prefect#6777]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

For our use case, we dynamically load modules based on the modules name. These modules can sometimes be installed as system packages and are not relative to the current directory for creating a deployment.

Introduced in 2.3.2, when attempting to use a module that is not relative to the current directory, Prefect fails with a value error.

### Reproduction

```python
module, method = module_name.rsplit('.', 1)
        mod = importlib.import_module(module)
        workflow_flow = getattr(mod, method)

        deployment = Deployment.build_from_flow(
            flow=workflow_flow,
            name=schedule.id,
            version=1,
            work_queue_name="default",
            schedule=deployment_schedule,
        )
        deployment.apply()
```
```


### Error

```
deployment = Deployment.build_from_flow(
--
File "/opt/venv/lib/python3.8/site-packages/prefect/utilities/asyncutils.py", line 212, in wrapper
return run_async_in_new_loop(async_fn, *args, **kwargs)
File "/opt/venv/lib/python3.8/site-packages/prefect/utilities/asyncutils.py", line 141, in run_async_in_new_loop
return anyio.run(partial(__fn, *args, **kwargs))
File "/opt/venv/lib/python3.8/site-packages/anyio/_core/_eventloop.py", line 70, in run
return asynclib.run(func, *args, **backend_options)
File "/opt/venv/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 292, in run
return native_run(wrapper(), debug=debug)
File "/usr/local/lib/python3.8/asyncio/runners.py", line 44, in run
return loop.run_until_complete(main)
File "/usr/local/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
return future.result()
File "/opt/venv/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
return await func(*args)
File "/opt/venv/lib/python3.8/site-packages/prefect/deployments.py", line 570, in build_from_flow
entry_path = Path(flow_file).absolute().relative_to(Path(".").absolute())
File "/usr/local/lib/python3.8/pathlib.py", line 908, in relative_to
raise ValueError("{!r} does not start with {!r}"
ValueError: '/opt/venv/lib/python3.8/site-packages/workflow_etl/flows/flow.py' does not start with '/opt/app'
```


### Versions

```

2.3.2

```


### Additional context

This change appears introduced here: https://github.com/PrefectHQ/prefect/pull/6695

**Comment by ColeMurray** [comment]
Hi all,

Following up here. Do we have a timeline on completion of this? It is blocking our path to upgrading past version 2.2.
From the previous commits, it looks like the logical change would be to remove introduction of "relative_to", but I am unsure what other parts of Prefect depend on this.

**Comment by madkinsz** [comment]
Hi! I'm happy to review a pull request for this but don't have the time to work on it right this second. Are you interested in contributing a fix?

**Comment by ColeMurray** [comment]
@madkinsz, 

Yes, happy to contribute a fix. I am a bit unsure of what the "correct" fix is here, as it looks like the relative lookup was intentionally added in https://github.com/PrefectHQ/prefect/pull/6554 https://github.com/PrefectHQ/prefect/pull/6695 to merge CLI and python behavior. Removing this behavior could have breaking effects on customers who have come to rely on this behavior.

A potential (albeit ugly) fix would be to try catch the value error thrown for a non-relative path, and use the previous behavior: https://github.com/PrefectHQ/prefect/blob/2.2.0/src/prefect/deployments.py#L480

**Comment by Trymzet** [comment]
This is also blocking me from using the official Prefect release, I use a fork with this change:
`entry_path = Path(flow_file).absolute().relative_to(Path(".").absolute())` -> `entry_path = Path(flow_file).absolute()` in `deployments.py`.

**Comment by zhen0** [comment]
Should be improved by https://github.com/PrefectHQ/prefect/pull/8930

**Comment by zealseeker** [comment]
I have the same problem. The flow (`main_flow`) is defined in an in-house package, which has been installed in a docker image. Then I use the following script.

```py
from mypackage.myflow import main_flow
deploy(main_flow.to_deployment("aa")
```
And it outputs:

'/usr/local/lib/python3.10/site-packages/mypakcage/myflow.py' is not in the subpath of '/builds/mypackage/'

Prefect version 3.2.9
